udata/core/spatial/search.py

```diff
@@ -43,15 +46,18 @@ class Meta:
                               payloads=True)
 
     @classmethod
-    def compute_weight(cls, population):
-        """Weight must be in the interval [0..2147483647]"""
-        if 0 <= population <= 2147483647:
-            return population
-        else:
-            if population < 0:  # country/eh population is -99.
-                return 0
-            else:  # World population is 6772425850.
-                return 2147483647
+    def compute_weight(cls, zone):
+        '''
+        Give a weight to the zone according its administrative level first
+        and then its population.
+        '''
+        # Each level give give a step
+        level = min(admin_levels.get(zone.level, ADMIN_LEVEL_MAX), 1)
+        level_weight = (ADMIN_LEVEL_MAX / level) * 10 * PONDERATION_STEP
+        # Population gives 0 < weight < 1000 to rank between level steps only
+        population = min(max(0, zone.population), POPULATION_MAX)
+        population_weight = (population / POPULATION_MAX) * PONDERATION_STEP
```

Comment:
Mmh ok got it for the ponderation, in that case why not below directly to the sum?